include/cuco/dynamic_map.cuh

```diff
@@ -32,6 +32,146 @@
 
 namespace cuco {
 
+namespace experimental {
+/**
+ * @brief A GPU-accelerated, unordered, associative container of key-value
+ * pairs with unique keys.
+ *
+ * This container automatically grows its capacity as necessary until device memory runs out.
+ *
+ * @tparam Key The type of the keys.
+ * @tparam T The type of the mapped values.
+ * @tparam Extent The type representing the extent of the container.
+ * @tparam Scope The thread scope for the container's operations.
+ * @tparam KeyEqual The equality comparison function for keys.
+ * @tparam ProbingScheme The probing scheme for resolving hash collisions.
+ * @tparam Allocator The allocator used for memory management.
+ * @tparam Storage The storage policy for the container.
+ */
+template <class Key,
+          class T,
+          class Extent             = cuco::extent<std::size_t>,
+          cuda::thread_scope Scope = cuda::thread_scope_device,
+          class KeyEqual           = thrust::equal_to<Key>,
+          class ProbingScheme      = cuco::linear_probing<4,  // CG size
+                                                          cuco::default_hash_function<Key>>,
+          class Allocator          = cuco::cuda_allocator<cuco::pair<Key, T>>,
+          class Storage            = cuco::storage<1>>
+class dynamic_map {
+  using map_type = static_map<Key, T, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>;
```

Comment:
I believe we should be using `T` and not `cuco::pair` for the value type. If you pass in `cuco::pair<Key, T>` it will exceed 8 bytes but more importantly `static_map::impl_type` already is parameterized over `open_addressing_impl` with `cuco::pair<Key, T>`